From e78aedbe5ddb6c56f780e233cd0244f51924adb0 Mon Sep 17 00:00:00 2001
From: Vitaly Margolin <vmargolin@habana.ai>
Date: Mon, 26 May 2025 07:43:45 +0300
Subject: [PATCH] [SW-224740] g3: lfs: refactor lfs logging

Refactor LFS logging to simplify and eliminate
some possible problems

Change-Id: I87122d68fd0d12161b40bc9a56ff5a179c37511f
Signed-off-by: Vitaly Margolin <vmargolin@habana.ai>
---

diff --git a/boards/hl/gaudi3/hl_gaudi3_agent_asic_defconfig b/boards/hl/gaudi3/hl_gaudi3_agent_asic_defconfig
index ffc5c8d..22bbaf4 100644
--- a/boards/hl/gaudi3/hl_gaudi3_agent_asic_defconfig
+++ b/boards/hl/gaudi3/hl_gaudi3_agent_asic_defconfig
@@ -107,11 +107,10 @@
 CONFIG_UART_CONSOLE_DEBUG_SERVER_HOOKS=y
 CONFIG_LFS_ENABLED=y
 CONFIG_LFS_OVER_SPI=y
-CONFIG_LFS_SHELL=n
+CONFIG_LFS_SHELL=y
 CONFIG_SHELL_HELP=n
 CONFIG_LFS_LOG_ENABLED_MANUALLY=n
 CONFIG_LFS_FOLDERS_SUPPORT=n
-CONFIG_LFS_LOG_FILE_SIZE=32768
-CONFIG_LFS_LOG_FILES_LIMIT=10
+CONFIG_LFS_LOG_FILE_SIZE=524288
 CONFIG_LFS_FORMAT_IF_MOUNT_FAILED=y
 CONFIG_ASSERT_VERBOSE=y
diff --git a/boards/hl/gaudi3/hl_gaudi3_mgmt_asic_defconfig b/boards/hl/gaudi3/hl_gaudi3_mgmt_asic_defconfig
index a0f1092..30b250b 100644
--- a/boards/hl/gaudi3/hl_gaudi3_mgmt_asic_defconfig
+++ b/boards/hl/gaudi3/hl_gaudi3_mgmt_asic_defconfig
@@ -160,8 +160,7 @@
 CONFIG_SHELL_HELP=y
 CONFIG_LFS_LOG_ENABLED_MANUALLY=n
 CONFIG_LFS_FOLDERS_SUPPORT=n
-CONFIG_LFS_LOG_FILE_SIZE=32768
-CONFIG_LFS_LOG_FILES_LIMIT=10
+CONFIG_LFS_LOG_FILE_SIZE=524288
 CONFIG_LFS_FORMAT_IF_MOUNT_FAILED=n
 CONFIG_ASSERT_VERBOSE=n
 CONFIG_PINCTRL=y
diff --git a/boards/hl/gaudi3/hl_gaudi3_preboot_asic_defconfig b/boards/hl/gaudi3/hl_gaudi3_preboot_asic_defconfig
index f5e9735..5ecc1cd 100644
--- a/boards/hl/gaudi3/hl_gaudi3_preboot_asic_defconfig
+++ b/boards/hl/gaudi3/hl_gaudi3_preboot_asic_defconfig
@@ -158,8 +158,7 @@
 CONFIG_SHELL_HELP=y
 CONFIG_LFS_LOG_ENABLED_MANUALLY=n
 CONFIG_LFS_FOLDERS_SUPPORT=n
-CONFIG_LFS_LOG_FILE_SIZE=32768
-CONFIG_LFS_LOG_FILES_LIMIT=10
+CONFIG_LFS_LOG_FILE_SIZE=524288
 CONFIG_LFS_FORMAT_IF_MOUNT_FAILED=n
 CONFIG_ASSERT_VERBOSE=y
 CONFIG_PINCTRL=y
diff --git a/drivers/misc/hl_npu/gaudi3/hl.c b/drivers/misc/hl_npu/gaudi3/hl.c
index 8bb6997..094da25 100644
--- a/drivers/misc/hl_npu/gaudi3/hl.c
+++ b/drivers/misc/hl_npu/gaudi3/hl.c
@@ -74,6 +74,10 @@
 #include <telemetry_ops.h>
 #include <hl_devs.h>
 
+#ifdef LFS_LOG_ENABLED
+#include "lfs_zephyr_init.h"
+#endif
+
 #ifndef TARGET_IS_ONIX
 #include <hl_arc_cfg.h>
 #endif
@@ -377,11 +381,21 @@
 		i2c_shutdown(dev, 0);
 }
 
+#ifdef LFS_LOG_ENABLED
+static void shutdown_lfs_log(void)
+{
+	module_lfs_shutdown();
+}
+#else
+static void shutdown_lfs_log(void) {}
+#endif
+
 static void hl_reset_handler(struct k_work *work)
 {
 	const struct device *dev = DEVICE_DT_INST_GET(0);
 	struct hl_drv_data *data = DEV_DATA(dev);
 
+	shutdown_lfs_log();
 	telem_enable(false);
 	hl_prepare_hard_rst(data);
 	hl_gaudi3_store_ecc_persistent_data(dev);
diff --git a/modules/Kconfig.lfs b/modules/Kconfig.lfs
index dd0d219..35a954f 100644
--- a/modules/Kconfig.lfs
+++ b/modules/Kconfig.lfs
@@ -28,10 +28,6 @@
 	int "Size of single LittleFS log file in bytes"
 	default 32768
 
-config LFS_LOG_FILES_LIMIT
-	int "Max. number of each LittleFS log file"
-	default 10
-
 config LFS_FORMAT_IF_MOUNT_FAILED
 	bool "Format LFS partition, if mount failed"
 	default n
